portable/NetworkInterface/xilinx_ultrascale/NetworkInterface.c

```diff
@@ -90,13 +90,6 @@
     #define iptraceEMAC_TASK_STARTING()    do {} while( 0 )
 #endif
 
-/* Default the size of the stack used by the EMAC deferred handler task to twice
- * the size of the stack used by the idle task - but allow this to be overridden in
- * FreeRTOSConfig.h as configMINIMAL_STACK_SIZE is a user definable constant. */
-#ifndef configEMAC_TASK_STACK_SIZE
-    #define configEMAC_TASK_STACK_SIZE    ( 8 * configMINIMAL_STACK_SIZE )
```

Comment:
Yes Aniruddha, you are right, the default stack size for the EMAC task is not the same on every platform. I just looked it up:
~~~
  ATSAM4E            :  4 * configMINIMAL_STACK_SIZE
  DriverSAM          :  4 * configMINIMAL_STACK_SIZE
  ksz8851snl         :  6 * configMINIMAL_STACK_SIZE
  M487               :  2 * configMINIMAL_STACK_SIZE
  STM32Fxx           :  2 * configMINIMAL_STACK_SIZE
  xilinx_ultrascale  :  8 * configMINIMAL_STACK_SIZE
  Zynq               :  8 * configMINIMAL_STACK_SIZE
~~~
That make it impossible to define the default `ipconfigEMAC_TASK_STACK_SIZE_WORDS` in `FreeRTOSIPConfigDefaults.h` that works for all platforms.

Do you have a suggestion?

We could set the default at `8 * configMINIMAL_STACK_SIZE`. That will work unless the platform has very little RAM available.

Or we could place the following declaration in each of the 7 versions of NetworkInterface.c:
~~~c
#ifndef ipconfigEMAC_TASK_STACK_SIZE_WORDS
    /* Check if the earlier name of the macro is still defined. */
    #ifdef configEMAC_TASK_STACK_SIZE
        #pragma message ("This macro is now called ipconfigEMAC_TASK_STACK_SIZE_WORDS")
        #define ipconfigEMAC_TASK_STACK_SIZE_WORDS    ( configEMAC_TASK_STACK_SIZE )
    #else
        #define ipconfigEMAC_TASK_STACK_SIZE_WORDS    ( XXX * configMINIMAL_STACK_SIZE )
        /* where XXX equals 2, 4, 6 or 8. */
    #endif
#endif
~~~
What do you think?

Reply:
I see.
I was wondering why is there so much disparity? Can all of the tasks not work with `4*configMINIMAL_STACK_SIZE`?
Maybe the 8x is overprovisioned?

But, regardless of why that is, I am not in favor of making a default value - since default might not work for a new driver which we might want to add later.

Maybe, it should be EMAC specific with no default value. If that is not defined, then we'll just print an error. This would warrant naming it something like `emacTASK_STACK_SIZE`.

What do you think @htibosch ?